File: Algotherms/Pattern Searching/Naive Searching/NaiveSearch.py
def NaiveSearch(text, pattern):
    text_size = len(text)
    pattern_size = len(pattern)

    if text_size < pattern_size:
        print("Pattern not found")
        return
    for i in range(text_size - pattern_size + 1):
        j = 0
        while j < pattern_size:
            if (text[i + j] != pattern[j]):
                break
            j += 1
        if j == pattern_size:
            print("Pattern found at index: ", i)

text = "THIS IS A TEST TEXT"
pattern = "TEST"
NaiveSearch(text, pattern)

File: Algotherms/Pattern Searching/Naive Searching/NaiveSearch.java
public class NaiveSearch {
    public static void main(String[] args) {
        String text = "THIS IS A TEST TEXT";
        String pattern = "TEST";
        naive(text, pattern);
    }

    public static void naive(String text, String pattern) {
        int text_size = text.length();
        int pattern_size = pattern.length();
        int j;

        for (int i = 0; i < text_size - pattern_size; i++) {
            for (j = 0; j < pattern_size; j++) {
                if (text.charAt(i + j) != pattern.charAt(j)) {
                    break;
                }
            }
            if (j == pattern_size) {
                System.out.println("Pattern found at index " + i);
            }
        }
    }
}
